crashtracker/src/collector/api.rs

```diff
@@ -593,3 +593,158 @@ fn test_altstack_nouse() -> anyhow::Result<()> {
+    //           if (pid == -1) {
+    //               int e = errno;
+    //               if (e == ECHILD)
+    //                   break;
+    //               rb_syserr_fail(e, 0);
+    //           }
+    //           rb_ary_push(result, rb_assoc_new(PIDT2NUM(pid), rb_last_status_get()));
+    //       }
+    //       return result;
+    //   }
+    //
+    // The intent here is to wait for all of one's child processes to exit.  This is a pretty
+    // standard operation in multi-process situations, with one important caveat:  usually you know
+    // your children ahead of time and can wait on them in a controlled, intentional matter.
+    // Previous versions of crashtracking, which spawned long-lived receiver processes, would
+    // interfere with this
+    //
+    // This implements the inner behavior of a test which allows the caller to control which
+    // options are used.
+    use crate::StacktraceCollection;
+    use chrono::Utc;
+    use ddcommon::Endpoint;
+
+    let time = Utc::now().to_rfc3339();
+    let dir = "/tmp/crashreports/";
+    let output_url = format!("file://{dir}{time}.txt");
+
+    let endpoint = Some(Endpoint::from_slice(&output_url));
+
+    let path_to_receiver_binary =
+        "/tmp/libdatadog/bin/libdatadog-crashtracking-receiver".to_string();
+    let create_alt_stack = false;
+    let use_alt_stack = false;
+    let resolve_frames = StacktraceCollection::EnabledWithInprocessSymbols;
+    let stderr_filename = Some(format!("{dir}/stderr_{time}.txt"));
+    let stdout_filename = Some(format!("{dir}/stdout_{time}.txt"));
+    let timeout_ms = 10_000;
+    let receiver_config = CrashtrackerReceiverConfig::new(
+        vec![],
+        vec![],
+        path_to_receiver_binary,
+        stderr_filename,
+        stdout_filename,
+    )?;
+    let config = CrashtrackerConfiguration::new(
+        vec![],
+        create_alt_stack,
+        use_alt_stack,
+        endpoint,
+        resolve_frames,
+        timeout_ms,
+    )?;
+
+    let metadata = CrashtrackerMetadata::new(
+        "libname".to_string(),
+        "version".to_string(),
+        "family".to_string(),
+        vec![],
+    );
+
+    // Since this test ultimately mutates process state, it's done inside of a fork just like the
+    // other tests of the same ilk.
+    match unsafe { libc::fork() } {
+        -1 => {
+            panic!("Failed to fork");
+        }
+        0 => {
+            // Child process
+            // This is where the test actually happens!
+            init(config, receiver_config, metadata)?;
+
+            // Now spawn some short-lived child processes.
+            // Note:  it's easy to confirm this test actually works by cranking the sleep duration
+            // up past the timeout duration. At such a point, the test should fail.
+            let mut children = vec![];
+            let sleep_duration_ms = 100;
+            let timeout_duration_ms = 150;
+            for _ in 0..10 {
```

Comment:
No reason.  Forks should spawn pretty quickly and my favorite random number, 17, subjectively felt too large.

In fact this test could be done with just a single fork, but since the `waitall()` pattern is typically done after a fan-out operation, I felt like it needed a number of children bigger than three.